### PR TITLE
check whether hotels in the filterArray are in the overall list before returning results

### PR DESCRIPTION
**Pull request by owennicol**
#### What are the links to relevant tickets?
https://hxshortbreaks.atlassian.net/browse/WEB-7869

#### What does this PR do?
This fixes a bug where, if the hotels in the 'filterArray' are not present in the hotels list, it displayed all hotels as 'featured' as well as all hotels as not featured.

#### What functional/unit tests does this PR have?
no additional

#### Screenshots (if appropriate)
---
#### Quality checklist (for PR author to complete BEFORE code review):
- [ ] I've checked there is appropriate unit test coverage.
- [x] I've updated documentation with any changes to procedures.
- [x] I've checked this work against the requirements of the Jira.
- [x] This change passes all necessary tests (cross browser, automated or otherwise).
- [x] I am ready for this to be code reviewed, merged and tested.

#### Reviewer 1 @reubensearle
- [x] I agree with the assumptions made in the quality checklist.
- [x] I’ve witnessed the work behaving as expected.
- [x] I’ve checked for appropriate test coverage.
- [x] I’ve checked for coding anti-patterns.
- [x] I've checked this work against the requirements of the Jira.
- [x] I don’t believe this work introduces unnecessary technical debt.
- [x] +1 from me!

#### Reviewer 2 @taijuten
- [x] I agree with the assumptions made in the quality checklist.
- [x] I’ve witnessed the work behaving as expected.
- [x] I’ve checked for appropriate test coverage.
- [x] I’ve checked for coding anti-patterns.
- [x] I've checked this work against the requirements of the Jira.
- [x] I don’t believe this work introduces unnecessary technical debt.
- [x] +1 from me!

### CODE REVIEW

File: helpers/sortAndFilterHotels.js [hunk 1]
```diff
@@ -17,27 +16,48 @@ module.exports = function(dust) {
...
+      .map( function( object ) {
+        // get the IDs of all hotels in the list
+        return object.links.hotelProducts.ids[ 0 ];
+      } )
+      .filter( function( hotelID ) {
+        // only return hotels with the ID of those in the filterArray
+        return filterArray.indexOf( hotelID ) > -1;
+      } )
+      //count
+      .length;
```
Comment (taijuten) [inline helpers/sortAndFilterHotels.js]: is this variable name correct, if it's just a length?

**Comment by taijuten** [comment]
Spoken to Owen about requiring a Wall PR for this too, to allow for easy testing locally

File: helpers/sortAndFilterHotels.js (hunk 1 above)
Comment (owennicol) [inline helpers/sortAndFilterHotels.js]: Where I'm using it, I wanted it to be explicit that I"m checking for the existence of the hotels in the list

File: helpers/sortAndFilterHotels.js [hunk 2]
```diff
@@ -17,27 +16,48 @@ module.exports = function(dust) {
    * @example {@_sortAndFilterHotels parent=packageRatesReply.packageRates node="grossPrice" filterArray=_brandConfig.featuredHotel filterValue="x.links.hotelProducts.ids[0]" include="true"} {/_sortAndFilterHotels} output loop of parent object sorted by node and filtered by filterArray
    */
 
-  dust.helpers._sortAndFilterHotels = function(chunk, context, bodies, params) {
+  dust.helpers._sortAndFilterHotels = function( chunk, context, bodies, params ) {
     var filterArray = params.filterArray;
```
Comment (ReubenSearle) [inline helpers/sortAndFilterHotels.js]: Should we check for an undefined filterArray? Not currently handling that.

Comment (owennicol) [inline helpers/sortAndFilterHotels.js]: This has beem implemented

**Pull request merged**